source/server/converters.h

```diff
@@ -317,6 +318,19 @@ typename TypeToStorageType<TDriverType, TGrpcType>::StorageType allocate_output_
 }
 
 }  // namespace converters
+
+const int kMaxGrpcErrorDescriptionSize = 4096;
+
+inline ::grpc::Status ApiErrorAndDescriptionToStatus(google::protobuf::int32 status, std::string& description)
+{
+    converters::trim_trailing_nulls(description);
+    // TODO: May need full-fledged json escape logic.
+    for (size_t index = 0; (index = description.find('"', index)) != std::string::npos; ++index) {
+        description.replace(index, 1, "\\\"");
+    }
+    std::string errorMessage = "{ \"code\":\"" + std::to_string(status) + "\", \"" + description + "\" }";
```

Comment:
Does this produce an invalid JSON object? I don't see an object key for `description`

```
{ "code":"-50150", "An unexpected error occurred..." }
```

Reply:
Sigh. `quality(spec) > quality(implementation)` 😬